### ISSUE TITLE
Implementation of a "heartbeat".

### ISSUE THREAD

**Issue by firew4lker** [beckdac/ESP8266-transparent-bridge#13]
Since the code can't interpret AT command originating from the UART, what is the best way for a microcontroller to check if the module is alive or frozen?

Maybe a mode that GPIO -2 toggles every couple o seconds or so?

The best scenario would be UART AT commands support. Is it even possible to implement support of AT command from the UART side?

**Comment by beckdac** [comment]
I see, so the unit attached to the ESP8266 wants to know if a destination is available?

**Comment by firew4lker** [comment]
No, if the ESP8266 is ok. Eg. if it has frozen the microcontroller to be able to reset it.

**Comment by beckdac** [comment]
OK.  How does the ESP8266 monitor the microcontroller?  Lack of serial activity then it throws the GPIO pin?

**Comment by firew4lker** [comment]
I mean the opposite. The microcontroller to monitor the ESP8266. If for example the ESP8266 stops toggling a GPIO then the mcu will know that the module is stuck and should be restarted.

I have notice that it fairly common for the "ebay tyoe" modules to freeze.

**Comment by beckdac** [comment]
OK.
